JobConfig/common/mu2eg4.fcl

```diff
@@ -22,13 +22,168 @@ mu2e.physics.producers.g4run.commonBase: {
    Mu2eG4CommonCut: {}
 }
 
+
+
+#--------Beging of cosmic part------
+
+crvOuterBox: {
+   type: union
+   pars:
+   [
+   // Upstream plane
+   { type: plane normal: [  0, 0, -1 ] point : [ 0, 0, -3400 ] },
+   // Downstream plane
+   { type: plane normal: [  0, 0, 1 ] point : [ 0, 0, 19800 ] },
+   // +x plane
+   { type: plane normal: [  1, 0, 0 ] point : [ 1500, 0, 0 ] },
+   // -x plane (outside of cryo box)
+   { type: plane normal: [  -1, 0, 0 ] point : [ -8200, 0, 0 ] },
+   // +y plane
+   { type: plane normal: [  0, 1, 0 ] point : [ 0, 4000, 0 ] },
+   // -y plane
+   { type: plane normal: [  0, -1, 0 ] point : [ 0, -2100, 0 ] }
+   ]
+}
+
+crvInnerBox: {
+   type: intersection
+   pars:
+   [
+   // Upstream plane
+   { type: plane normal: [  0, 0, 1 ] point : [ 0, 0, -2200 ] },
+   // Downstream plane
+   { type: plane normal: [  0, 0, -1 ] point : [ 0, 0, 18500 ] },
+   // +x plane
+   { type: plane normal: [  -1, 0, 0 ] point : [ -1500, 0, 0 ] },
+   // -x plane
+   { type: plane normal: [  1, 0, 0 ] point : [ -6300, 0, 0 ] },
+   // +y plane
+   { type: plane normal: [  0, -1, 0 ] point : [ 0, 2500, 0 ] },
+   // -y plane
+   { type: plane normal: [  0, 1, 0 ] point : [ 0, -2100, 0 ] }
+   ]
+}
+
+crvStage1End: {
+   type: inVolume
+   pars: [DSOuterCryoShell, DSInnerCryoShell, DSUpEndWallShell, TS5OuterCryoShell, TS5CryoInsVac, TS5Vacuum, Coll51, DS1Vacuum, DS2Vacuum, DS3Vacuum]
+   write: crvStage1
+}
+
+cosmicKineticEnergyCutS1:
+{
+   type: union
+   pars:
+   [
+      {
+      type: intersection
+      pars:
+      [
+	{ type: union pars: [ @local::crvOuterBox, @local::crvInnerBox ] },
+      	{ type: kineticEnergy cut: 10.0 },
+      	{ type: pdgId pars: [ 22, 2112 ] }
+      ]
+      },
+      {
+      // 5 MeV can only make a hit in 1 CRV layer
+      type: intersection
+      pars:
+      [
+	{ type: union pars: [ @local::crvOuterBox, @local::crvInnerBox ] },
+        { type: kineticEnergy cut: 5.0 },
+        { type: pdgId pars: [ 2212, 11, -11 ] }
+      ]
+      }
+   ]
+}
+
+cosmicKineticEnergyCutS2: {
+// Kill lower momentum particles outside of the CRV box
+   type: intersection
+   pars:
+   [
+	{ type: intersection pars: [
+	{ type: kineticEnergy cut: 80.0 },
+ 	{ type: pdgId pars: [ 22, 11, -11, 2112 ] }]},
+ 	{ type: union pars: [ @local::crvOuterBox ] }
+   ]
+}
+
+Cosmic:{
```

Comment:
Cosmic block should be in JobConfig/cosmic/prolog.fcl.   mu2eg4 should be general G4 config used by multiple parts of JobConfig.

Reply:
I thought you wanted cosmic portion defined in mu2eg4.
I'll move everything to JobConfig/cosmic/prolog.fcl, and include it from primary/prolog.fcl
That's what I originally had in mind.